src/Database/Exceptions/DatabaseException.php

```diff
@@ -0,0 +1,11 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Database\Exceptions;
+
+use Exception;
+
+abstract class DatabaseException extends Exception
```

Comment:
While not required now, it could be useful for grouping in try-catches

Alternatively, this could be an interface that just extends Throwable, but idk class felt more natural here

Reply:
Well you can also group catches like so:

```php
catch(ExceptionA|ExceptionB|ExceptionC $exception)
```

So I think separate exceptions will be better

Reply:
Of course you can! But if you want to catch any problem with DB, especially as we add new ones, having a catch-all for db would be nice imho